From c308beadeb8f88cd866c8cbc2984d5b38e79eaa8 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Wed, 6 Jun 2018 14:02:43 +0800
Subject: [PATCH 3915/5242] MLK-17305 [MX8QXP-MEK] VPU: "couldn't set
 vpu_dec_clk clk rate to 600000000 (-22)" and
 "clk: couldn't set vpu_enc_clk clk rate to
 600000000 (-22), current rate: 0" when boot up.
 100%

commit  cfbc7684a188df62b8fe6b121cf8afb722814f1f from
https://source.codeaurora.org/external/imx/linux-imx.git

vpu clock is not settable, remove the assigned-clock-rates from the dts

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    2 --
 drivers/mxc/vpu-decoder-b0/vpu_b0.c            |    2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index a3cb4d6..43c0a41 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -3113,7 +3113,6 @@
 		clocks = <&clk IMX8QXP_VPU_DEC_CLK>;
 		clock-names = "vpu_clk";
 		assigned-clocks = <&clk IMX8QXP_VPU_DEC_CLK>;
-		assigned-clock-rates = <600000000>;
 		power-domains = <&pd_vpu_dec>;
 		status = "disabled";
 	};
@@ -3127,7 +3126,6 @@
 		clocks = <&clk IMX8QXP_VPU_ENC_CLK>;
 		clock-names = "vpu_encoder_clk";
 		assigned-clocks = <&clk IMX8QXP_VPU_ENC_CLK>;
-		assigned-clock-rates = <600000000>;
 		power-domains = <&pd_vpu_enc>;
 		status = "disabled";
 	};
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index 239e553..084235b 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -1814,6 +1814,8 @@ static void vpu_api_event_handler(struct vpu_ctx *ctx, u_int32 uStrIdx, u_int32
 	}	break;
 	case VID_API_EVENT_FIRMWARE_XCPT:
 		vpu_dbg(LVL_ERR, "warning: FIRMWARE hang, and send event VID_API_EVENT_FIRMWARE_XCPT\n");
+	case VID_API_EVENT_DEC_CFG_INFO:
+		break;
 	default:
 		vpu_dbg(LVL_ERR, "warning: uEvent %d is not handled\n", uEvent);
 		break;
-- 
1.7.9.5

